### PR TITLE
Adding CleanZookeeper utility to accumulo admin command

### PR DESCRIPTION
**Pull request by cshannon**
Found another utility that can be moved to the accumulo admin command. This utility cleans all old instances out of Zookeeper (except the current one). It's a bit different than the DeleteZooInstance command as that command allows deleting a single instance that is specified so it's worth having both.

This is a follow on to https://github.com/apache/accumulo/issues/2807

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/util/CleanZookeeper.java [hunk 1]
```diff
@@ -67,25 +48,26 @@ public static void main(String[] args) {
             String instanceNamePath = root + Constants.ZINSTANCES + "/" + instanceName;
             byte[] id = zk.getData(instanceNamePath);
             if (id != null && !new String(id, UTF_8).equals(context.getInstanceID().canonical())) {
-              try {
-                zk.recursiveDelete(instanceNamePath, NodeMissingPolicy.SKIP);
-              } catch (KeeperException.NoAuthException ex) {
-                log.warn("Unable to delete {}", instanceNamePath);
-              }
+              delete(zk, instanceNamePath);
+              System.out.println("Deleted instance " + instanceName);
```
Comment (dlmarion) [inline server/base/src/main/java/org/apache/accumulo/server/util/CleanZookeeper.java:52]:
```suggestion
              System.out.println("Deleted instance: " + instanceName);
```

**Review by dlmarion — APPROVED** [review]
Small nit, LGTM

File: server/base/src/main/java/org/apache/accumulo/server/util/CleanZookeeper.java (hunk 1 above)
Comment (cshannon) [reply server/base/src/main/java/org/apache/accumulo/server/util/CleanZookeeper.java:52]: Thanks for the review, I should be able to push that quick change tomorrow when I get a chance.

**Comment by cshannon** [comment]
@milleruntime - Any thoughts on this before merging since you looked at the others?

**Comment by milleruntime** [comment]
This looks good but #2914 is a priority and they both touch Admin.

**Comment by ctubbsii** [comment]
> Found another utility that can be moved to the accumulo admin command. This utility cleans all old instances out of Zookeeper (except the current one). It's a bit different than the DeleteZooInstance command as that command allows deleting a single instance that is specified so it's worth having both.

I'm not sure it's worth having both. The delete one could just have a flag to delete all non-current ones.

**Comment by cshannon** [comment]
> > Found another utility that can be moved to the accumulo admin command. This utility cleans all old instances out of Zookeeper (except the current one). It's a bit different than the DeleteZooInstance command as that command allows deleting a single instance that is specified so it's worth having both.
> 
> I'm not sure it's worth having both. The delete one could just have a flag to delete all non-current ones.

I like that idea, I can simply move this functionality to the other command pretty easily and have a flag as you suggested. Having 1 utility/command to clean up zookeeper instances is probably less confusing and I can just update the troubleshooting documentation.

If no one else has any other comments or objects I can create a new PR tomorrow (and close this one) that moves the functionality here into the other utility and then removes the CleanZookeeper utility as it won't be needed.

**Comment by cshannon** [comment]
Closing this PR out as I'm going to create a new one and just merge the functionality of CleanZookeeper into the DeleteZooInstance utility with a flag as suggested.

**Pull request closed without merging**